src/components/PostComment.jsx

```diff
@@ -0,0 +1,73 @@
+import propTypes from "prop-types";
+import { UserContext } from "../contexts/User";
+import { useContext, useState } from "react";
+import { postData } from "./api";
+import styles from "../css/PostComment.module.css";
+
+function PostComment({ article_id }) {
+  const { user } = useContext(UserContext);
+  const [postedComment, setPostedComment] = useState({
+    author: user,
+    body: "",
+  });
+  const [isButtonDisabled, setButtonDisabled] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  function handleSubmit(event) {
+    setIsSubmitted(false);
+    event.preventDefault();
+    postData(article_id, postedComment);
+    setButtonDisabled(true);
+    setPostedComment({ author: user, body: "" });
+    setIsSubmitted(true);
+  }
+
+  function handleChange(event) {
+    setPostedComment({
```

Comment:
use a callback within the setstate here to manage the asynchronous nature of the set state function. see this: https://www.dhiwise.com/post/guide-to-state-management-with-react-setstate-callback